playgame.py

```diff
@@ -24,6 +25,11 @@ def main(argv):
     else:
         run_one_game(args)
 
+def ensure_path(path):
+    directories_from_path = "/".join(path.split("/")[:-1])
```

Comment:
Code smell:
Joining '/' to the path adds a directory at the root of your system, not the root of the project. Also, '/' is not the only correct path separator. This will break for non-UNIX systems (Windows uses '\', for instance)

The directory part of a file path can be separated with `os.path.dirname`, the file part with `os.path.basename`. As long as `path` includes a filename (which we can expect) we do not need to manage paths ourselves [https://docs.python.org/2/library/os.path.html#os.path.dirname](url)

```suggestion
    dir_name = os.path.dirname(path)
```

Reply:
If you've been running sample.sh and tournament_sample.sh to test, you probably have 'results' folders in the root of your system and you can delete them